classes/class-wpcom-liveblog-entry.php

```diff
@@ -535,6 +535,38 @@ public static function get_entry_title( $entry ) {
 		return wp_trim_words( $entry->content, 10, '…' );
 	}
 
+	/**
+	 * Get Entry first image if present.
+	 *
+	 * @param  object $entry Entry.
+	 * @return string        Image src.
+	 */
+	public static function get_entry_first_image( $entry ) {
+		$entry_id = ( $entry instanceof WPCOM_Liveblog_Entry ) ? $entry->get_id() : $entry->id;
+		$entry_content = ( $entry instanceof WPCOM_Liveblog_Entry ) ? $entry->get_content() : $entry->content;
+		$key    = 'liveblog_entry_' . $entry_id . 'first_image';
+		$cached = wp_cache_get( $key, 'liveblog' );
```

Comment:
We'll need to invalidate this cache if the entry is updated or deleted.

Reply:
Update/delete entries are stored as new comments, and have their own `id`s, so there isn't a need to invalidate as the data would never be stale.  Granted, once an entry is updated, we could delete the old cache entry and prime the cache with the updated entry, but it's not necessary.  Provided the above, do you have a recommendation on what you'd like pursued?

Reply:
Wow, that's super-weird. Given that, we don't need to do anything.